doc/source/tutorial/interpolate.rst

```diff
@@ -10,38 +10,494 @@ Interpolation (:mod:`scipy.interpolate`)
+    >>> ax[1].plot(xnew, spl(xnew, nu=1), '--', label='1st derivative')
+    >>> ax[2].plot(xnew, spl(xnew, nu=2), '--', label='2nd derivative')
+    >>> ax[3].plot(xnew, spl(xnew, nu=3), '--', label='3rd derivative')
+    >>> for j in range(4):
+    ...     ax[j].legend(loc='best')
+    >>> plt.tight_layout()
+    >>> plt.show()
+
+Note that the first and second derivatives are continuous by construction, and
+the third derivative jumps at data points. 
+
+
+Monotone interpolants
+---------------------
+
+Cubic splines are by construction twice continuously differentiable. This may
+lead to the spline function oscillating and ''overshooting'' in between the
+data points. In these situations, an alternative is to use the so-called
+*monotone* cubic interpolants: these are constructed to be only once
+continuously differentiable, and attempt to preserve the local shape implied
+by the data. There are two objects of this class in `scipy.interpolate` :
+`PchipInterpolator` and `Akima1DInterpolator` . To illustrate, let's consider
+a data with an outlier:
+
+.. plot::
+
+    >>> from scipy.interpolate import CubicSpline, PchipInterpolator, Akima1DInterpolator
+    >>> x = np.array([1., 2., 3., 4., 4.5, 5., 6., 7., 8])
+    >>> y = x**2
+    >>> y[4] += 101
+
+    >>> import matplotlib.pyplot as plt
+    >>> xx = np.linspace(1, 8, 51)
+    >>> plt.plot(xx, CubicSpline(x, y)(xx), '--', label='spline')
+    >>> plt.plot(xx, Akima1DInterpolator(x, y)(xx), '-', label='Akima1D')
+    >>> plt.plot(xx, PchipInterpolator(x, y)(xx), '-', label='pchip')
+    >>> plt.plot(x, y, 'o')
+    >>> plt.legend()
+    >>> plt.show()
+
+
+Piecewise polynomials
+---------------------
+
+Internally, `CubicSpline` and monotone interpolants are represented as instances
+of a `PPoly` class, which represents pieciwise polynomials in terms of
+breakpoints and coefficients (`PPoly` objects can represent polynomials of 
+arbitrary orders, not only cubics). For the data array ``x``, breakpoints are at
+the data points, and the array of coefficients, ``c`` , define cubic polynomials
+such that ``c[k, j]`` is a coefficient for ``(x - x[j])**(3-k)`` on the segment
+between ``x[j]`` and ``x[j+1]`` .
+
+
+.. _tutorial-interpolate_ppoly:
+
+Manipulating ``PPoly`` objects
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+`PPoly` objects have convenient methods for constructing derivatives
+and antiderivatives, computing integrals and root-finding. For example, we
+tabulate the sine function and find the roots of its derivative.
+
+    >>> from scipy.interpolate import CubicSpline
+    >>> x = np.linspace(0, 10, 71)
+    >>> y = np.sin(x)
+    >>> spl = CubicSpline(x, y)
+
+Now, differentiate the spline:
+
+    >>> dspl = spl.derivative()
+
+Here ``dspl`` is a `PPoly` instance which represents a polynomial approximation
+to the derivative of the original object, ``spl`` . Evaluating ``dspl`` at a
+fixed argument is equivalent to evaluating the original spline with the ``nu=1``
```

Comment:
Not sure what you mean, mind expanding? If you have a preference on how to structure / explain this, I'd go with it. 
Or maybe it's a sign of this whole subsection being extraneous?

Reply:
In the part above:
```
A `CubicSpline` object's ``__call__`` method accepts both scalar values and
arrays. It also accepts a second argument, ``nu``, to evaluate the 
derivative of order ``nu``. As an example, we plot the derivatives of a spline:
```

I was surprised that you could use the `__call__` method and pass a cryptically-named keyword argument to evaluate the derivative of the spline. It doesn't seem like as good API design as this object-oriented `derivative` method here, so I think we should emphasize the use of the `derivative` method instead. The suggestion is to eliminate the discussion of the `nu` keyward argument of `__call__`, and just talk about derivatives down here.

Reply:
Well, I respectfully disagree. Without going into a discussion of what is and what is not a good design, I'd like to somehow demonstrate both ways in the tutorial. The main difference is that `spl(x, nu=2)` evaluates the second derivative in-place, and `spl.derivative(2)` constructs a new object of a degree two smaller which represents the second derivative. I'm open to suggestions of how to tone down the use of the `nu` argument if you feel strongly about it, but some demonstration I'd like to keep.

Reply:
Oops, my word choice needed improvement there; sorry about that. I didn't notice the distinction before, so perhaps it's it can be emphasized more if the difference in performance is substantial.